docs/proposals/scheduling/average-replicas-assign/average-replicas-assign.md

```diff
@@ -0,0 +1,289 @@
+
+<!--
+在计算出每个保存每个 Cluster 的平均待分配实例数 avg_rep 之后，开始遍历被选出的 Cluster，并使用已经定义好的 TargetCluster 结构体定义分配结果。
+-->
+
+It is important to note that to ensure the allocated workloads can be successfully deployed, we also need to consider the current available resources of each member cluster. This can be obtained through the `karmada-scheduler-estimator`.
+
+<!--
+需要注意的是，为了保证被分配的负载能够被顺利地拉起，因此还需要考虑各个工作集群目前的可用资源，这可以通过 karmada-scheduler-estimator 获取。
+-->
+
+The specific allocation process is as follows:
+<!--
+具体的分配流程如下：
+-->
+1. Traverse all selected clusters to check how many replicas they can accommodate (i.e., AvailableReplica), then perform the following **one round** of traversal:
+   <!--
+   1.遍历所有被选出的 Cluster，查看其能容纳多少 replica（也就是 AvailableReplica），然后进行如下一轮遍历：
+   -->
+   * If AvailableReplica > the avg_rep corresponding to that cluster, allocate avg_rep instances to that cluster, and set the cluster's AvailableReplica to AvailableReplica - avg_rep. Since this cluster has not reached its full allocation, add it to the **unFullCluster slice (a global variable not bound to any specific cluster)** representing under-allocated clusters.
+   <!--
+   a.如果 AvailableReplica > 该 Cluster 对应的 avg_rep，那么给该 Cluster 分配 avg_rep 个实例，并将该 Cluster 的 AvailableReplica 设置为 AvailableReplica - avg_rep，由于该 Cluster 还没有分配满额，因此将其存入表示未满额集群的 unFullCluster（这是一个全局变量，不与任何一个 Cluster 绑定）切片中。
+   -->
+   * If AvailableReplica <= the avg_rep corresponding to that cluster, allocate the maximum AvailableReplica instances to that cluster, and add the remaining pending replicas (avg_rep - AvailableReplica) to the variable **sumPendingReplica (a global variable not bound to any specific cluster)**, which is initialized to zero.
+   <!--
+   b.如果 AvailableReplica <= 该 Cluster 对应的 avg_rep，那么给该 Cluster 分配满额的 AvailableReplica 个实例，同时将剩余的待分配 pendingReplica，也就是 avg_rep - AvailableReplica，添加到变量 sumPendingReplica（这是一个全局变量，不与任何一个 Cluster 绑定）中，该变量初始时为零。
+   -->
+2. After completing the first round of traversal, first sort the unFullCluster by Cluster.Score, then start traversing the unFullCluster. For each cluster in it, allocate one replica, and decrement both the cluster's AvailableReplica and sumPendingReplica by 1. Then check if the cluster's AvailableReplica is zero, and if so, remove that cluster from the unFullCluster.
+<!--
+2.在一轮遍历结束之后，首先对 unFullCluster 按照 Cluster.Score 进行排序，然后开始遍历 unFullCluster，对于其中的每个 Cluster，给其分配一个 replica，同时该集群的 AvailableReplica 和  sumPendingReplica 都自减 1，然后判断该 Cluster 的 AvailableReplica 是否为零，如果为零，则将该 Cluster 移出 unFullCluster。
+-->
+
+3. Repeat step 2 until sumPendingReplica is 0, completing the allocation.
+<!-- 
+3.循环第二步，直至 sumPendingReplica 为 0，分配结束。
+-->
+
+Since we have ensured in the Select phase that the total AvailableReplica of the selected clusters is greater than the number of replicas to be allocated, there is no need to worry about resource insufficiency.
```

Comment:
This requirement is met in the current **Select** phase. However, it does not mean that this requirement is always met in the future evolution. Therefore, the resource insufficiency still needs to be considered during the design.

Reply:
yeah, I have already pointed this out.